### DISCUSSION TITLE
Use a different sort strategy per command

### DISCUSSION THREAD

**Discussion by jotare** [minad/vertico#473] [Q&A] [answered]
Hi, I recently switched to vertico and I was wondering if it is possible to use a different sort function per command. 

As an example, I'd like to use a different sort function for `M-x` and `find-file` completions. I want to sort commands using history-alpha but sort files alphabetically (not by history).  

Thanks!

**Comment by minad** [comment, marked as answer]
https://github.com/minad/vertico?tab=readme-ov-file#configure-vertico-per-command-or-completion-category

**Reply by jotare** [reply]
Oh, sorry for the dumb question :facepalm: I didn't fully understand how that worked when I first read it...

Something like this works for me:
```elisp
(setq vertico-multiform-commands
  '((find-file (vertico-sort-function . vertico-sort-alpha))
     (execute-extended-command (vertico-sort-function . vertico-sort-history-alpha))
     (t (vertico-sort-function . vertico-sort-history-length-alpha))))
```

It keeps being a bit unclear for me which are the available categories you can configure with `vertico-multiform-categories`, not sure if it's explained somewhere. Anyway, feel free to close the discussion.

Thanks for your time!
